src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentReferenceCaptureLoweringPass.cs

```diff
@@ -33,32 +35,106 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument, DocumentInte
+                        }
+
+                        @class.Children.Insert(position, CreateField(rewritten.FieldTypeName, identifier));
+                    }
                 }
             }
         }
 
-        private IntermediateNode RewriteUsage(ClassDeclarationIntermediateNode classNode, IntermediateNode parent, TagHelperDirectiveAttributeIntermediateNode node)
+        private ReferenceCaptureIntermediateNode RewriteUsage(IntermediateNode parent, IntermediateToken identifier)
         {
-            // If we can't get a nonempty attribute name, do nothing because there will
-            // already be a diagnostic for empty values
-            var identifierToken = DetermineIdentifierToken(node);
-            if (identifierToken == null)
-            {
-                return node;
-            }
-
             // Determine whether this is an element capture or a component capture, and
             // if applicable the type name that will appear in the resulting capture code
             var componentTagHelper = (parent as ComponentIntermediateNode)?.Component;
             if (componentTagHelper != null)
             {
-                return new ReferenceCaptureIntermediateNode(identifierToken, componentTagHelper.GetTypeName());
+                return new ReferenceCaptureIntermediateNode(identifier, componentTagHelper.GetTypeName());
             }
             else
             {
-                return new ReferenceCaptureIntermediateNode(identifierToken);
+                return new ReferenceCaptureIntermediateNode(identifier);
+            }
+        }
+
+        private bool ShouldGenerateField(IntermediateNode parent)
+        {
+            var parameters = parent.FindDescendantNodes<TagHelperDirectiveAttributeParameterIntermediateNode>();
+            for (var j = 0; j < parameters.Count; j++)
+            {
+                var parameter = parameters[j];
+                if (parameter.TagHelper.IsRefTagHelper() && parameter.BoundAttributeParameter.Name == "suppressField")
+                {
+                    if (parameter.HasDiagnostics)
+                    {
+                        parent.Diagnostics.AddRange(parameter.GetAllDiagnostics());
+                    }
+
+                    parent.Children.Remove(parameter);
+
+                    if (parameter.AttributeStructure == AttributeStructure.Minimized)
+                    {
+                        return false;
+                    }
+
+                    // We do not support non-minimized attributes here because we can't allow the value to be dynamic.
+                    // As a design/experience decision, we don't let you write @ref:suppressField="false" even though
+                    // we could parse it. The rationale is that it's misleading, you type something that looks like code,
+                    // but it's not really.
+                    parent.Diagnostics.Add(ComponentDiagnosticFactory.Create_RefSuppressFieldNotMinimized(parameter.Source));
```

Comment:
Is there value in making this logic more generic if we want to enforce some other future bool directive attributes to have this error? In other words, do you envision needing this behavior for any other attributes in the future?

Reply:
I thought about that too since we'll probably have at least one more of these at some point (needs to be known at compile-time).

But still, YAGNI.... I don't want to run off and refactor my one usage of this pattern.